src/Symfony/Component/Routing/Generator/UrlGenerator.php

```diff
@@ -109,7 +113,14 @@ public function isStrictRequirements()
      */
     public function generate($name, $parameters = array(), $referenceType = self::ABSOLUTE_PATH)
     {
-        if (null === $route = $this->routes->get($name)) {
+        $locale = $parameters['_locale']
+            ?? $this->context->getParameter('_locale')
+            ?: $this->defaultLocale;
+        $route = $this->routes->get($name.'.'.$locale);
```

Comment:
Excuse my ignorance with the routing internals. But isnt this a BC break in some specific edge cases?

(I assume the UrlGenerator would get the default locale in the constructor automatically (ie Symfonys service definition is updated).)

If I have two routes with named `foo` and `foo.en`. They go to two different controllers. With current state of the PR, my second route will be overwritten by the "localized" version of `foo`.